code/modules/surgery/implant_removal.dm

```diff
@@ -1,81 +1,199 @@
-		if(case && !case.imp)
-			case.imp = implant
-			implant.forceMove(case)
-			case.update_appearance()
+			for(var/obj/item/implantcase/implant_case in range(1, user))
+				if(!implant_case.imp)
+					case = implant_case
+					break
+		if(!case)
+			to_chat(user, span_warning("You have no implant case in-hand or nearby to store extracted implants from [target]. Did you mean to destroy the implant?"))
+			return SURGERY_STEP_FAIL
+
+		var/chosen_implant = tgui_input_list(user, "Extract which implant?", "Surgery", sort_list(surgery.discovered_implants))
+		if(isnull(chosen_implant) || !(chosen_implant in target.implants))
+			return SURGERY_STEP_FAIL
+		if(user && target && user.Adjacent(target) && user.get_active_held_item() == tool && case && user.Adjacent(case) && !case.imp)
+			target_implant = chosen_implant
+			target_case = case
+			display_results(
+				user,
+				target,
+				span_notice("You begin to extract [chosen_implant] from [target]'s [parse_zone(target_zone)]..."),
+				span_notice("[user] begins to extract [chosen_implant] from [target]'s [parse_zone(target_zone)]."),
+				span_notice("[user] begins to extract something from [target]'s [parse_zone(target_zone)]."),
+			)
+			// I'm sorry, but you cannot tell what type of implants is being removed from sensation alone.
+			display_pain(target, "You can feel an implant being removed from your [parse_zone(target_zone)]!")
+		else
+			return SURGERY_STEP_FAIL
+
+	else
+		// We don't care about salvage, we just destroying the implant!
+		var/chosen_implant = tgui_input_list(user, "Destroy which implant?", "Surgery", sort_list(surgery.discovered_implants))
+		if(isnull(chosen_implant))
+			return SURGERY_STEP_FAIL
+		if(user && target && user.Adjacent(target) && user.get_active_held_item() == tool)
+			target_implant = chosen_implant
 			display_results(
 				user,
 				target,
-				span_notice("You place [implant] into [case]."),
-				span_notice("[user] places [implant] into [case]!"),
-				span_notice("[user] places it into [case]!"),
+				span_notice("You begin to destroy [chosen_implant] in [target]'s [parse_zone(target_zone)]..."),
+				span_notice("[user] begins to destroy [chosen_implant] in [target]'s [parse_zone(target_zone)]."),
+				span_notice("[user] begins to destroy something in [target]'s [parse_zone(target_zone)]."),
 			)
+			// I'm sorry, but you cannot tell what type of implants is being removed from sensation alone.
+			display_pain(target, "You can feel an implant being destroyed in your [parse_zone(target_zone)]!")
 		else
-			qdel(implant)
+			return SURGERY_STEP_FAIL
 
+/datum/surgery_step/destroy_extract_implant/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/implant_removal/surgery, default_display_results = FALSE)
+	if(!target_implant || !(target_implant in target.implants))
+		to_chat(user, span_warning("You can't find what you were looking for in [target]'s [target_zone]!"))
```

Comment:
```suggestion
	if(!target_implant || !(target_implant in target.implants))
		to_chat(user, span_warning("You can't find what you were looking for in [target]'s [target_zone]!"))
		return ..()
```

It feels like you can use early returns pretty extensively in this proc to reduce how far everything below is indented, might make the execution a little clearer.